config/options.go

```diff
@@ -441,58 +441,67 @@ func (o *Options) HTTPClientUserAgent() string {
 	return o.httpClientUserAgent
 }
 
+// KeyValueMap returns options as a map from string to string.
+func (o *Options) KeyValueMap() map[string]string {
+	keyValues := make(map[string]string)
```

Comment:
You could use a `map[string]interface{}` instead of having all these `sprintf` below.

Reply:
Returns string to string map intentionally.
This function majorly is used for output the debugging strings. If we do not print values here, we are going to print values later anyway.
I do not think the caller care about the type of values. (If they do, they should other functions of option)

Reply:
It's cleaner to print everything at the same place.